monkey/common/configuration/agent_configuration.py

```diff
@@ -24,6 +36,57 @@ class AgentConfiguration:
     payloads: List[PluginConfiguration]
     propagation: PropagationConfiguration
 
+    def __post_init__(self):
+        # This will raise an exception if the object is invalid. Calling this in __post__init()
+        # makes it impossible to construct an invalid object
+        try:
+            AgentConfigurationSchema().dump(self)
+        except Exception as err:
```

Comment:
Are we expecting anything but `MarshmallowError`?

Reply:
A better question is, "Is there any case where `AgentConfigurationSchema().dump()` could raise an exception for any reason other than an invalid exception?" To my knowledge, there's not.